test/maps/lua_testsuite.wmf/scripting/path.lua

```diff
@@ -22,12 +22,12 @@ function _array_contains(list, string)
 end
 
 function path_test:test_list_files()
-   local dirname = "tribes/buildings/productionsites/amazons/woodcutters_hut/"
-   assert_equal(3, #path.list_files(dirname .. "idle_?.png"))
-   assert_equal("tribes/buildings/productionsites/amazons/woodcutters_hut/idle_1.png", path.list_files(dirname .. "idle_?.png")[1])
-   assert_equal("tribes/buildings/productionsites/amazons/woodcutters_hut/idle_4.png", path.list_files(dirname .. "idle_?.png")[3])
+   local dirname = "tribes/buildings/productionsites/barbarians/barracks/"
+   assert_equal(4, #path.list_files(dirname .. "idle_*.png"))
```

Comment:
`path.list_files` can only detect files of the sort `basename_?? with a running number from 0 onwards – the system needed for the old multiple-files animations definitions but obsolete for spritesheets. The function is marked deprecated. This test could just be removed.

Reply:
Well, as far as I undrstood the code it should detect every file with the basename correctly matching the placeholder as an "f" placeholder does not match the 0.5 file it breaks. so with an "*" it should work, to detect the files.    
The test can be removed if we remove the lua function. If I can't get it to work with arrays i will simplify it for this branch as it is already to big a diff

Reply:
No it only matches this one case and does not work for files in general:

https://github.com/widelands/widelands/blob/408e06450c445d9d1af63929ba224f27ad72d7a7/src/scripting/lua_path.cc#L31-L79
https://github.com/widelands/widelands/blob/408e06450c445d9d1af63929ba224f27ad72d7a7/src/scripting/lua_path.cc#L148-L165

Reply:
ok so first I make it simple to get the tests go green. and we should remove all of the deprecated stuff afterwards.